lib/core/src/Cardano/Wallet/Primitive/CoinSelection/Random.hs

```diff
@@ -109,12 +107,12 @@ data TargetRange = TargetRange
 -- we set.
 -- @
 random
-    :: forall m e. MonadRandom m
+    :: forall e. ()
```

Comment:
Yes, testing was made a little harder because of this, plus, we also now loose the nice type signature that was nicely indicating which side-effects were needed. I might as well flag it as a technical debt to address later on and restore some purity in this.